lib/rack/request.rb

```diff
@@ -489,37 +584,24 @@ def parse_multipart
         Rack::Multipart.extract_multipart(self, query_parser)
       end
 
-      def split_ip_addresses(ip_addresses)
-        ip_addresses ? ip_addresses.strip.split(/[,\s]+/) : []
-      end
-
-      def hostname
-        if forwarded = get_header(HTTP_X_FORWARDED_HOST)
-          forwarded.split(/,\s?/).last
-        else
-          get_header(HTTP_HOST) ||
-            get_header(SERVER_NAME) ||
-            get_header(SERVER_ADDR)
-        end
+      def split_header(value)
```

Comment:
A generic name like `split_header` implies it will split all headers, which I don't think is possible as headers have different formats.  Maybe `split_ip_addresses` wasn't accurate, but I think `split_header` is more misleading as a name.